### DISCUSSION TITLE
Convert multi frame DICOM file to single frame DICOM files

### DISCUSSION THREAD

**Discussion by barna21** [fo-dicom/fo-dicom#1255] [Q&A] [answered]
Hello,

How can I convert multi fram DICOM file to standard single frame DICOM files?

any help is appreciated
thanks

**Comment by mrbean-bremen** [comment]
It depends on the type of multi-frame DICOM file you have. If this is some of the EnhancedXX storage classes, like EnhancedCT  or EnhancedMR, you basically have to create new datasets (`DicomDataset`) where you copy most of the shared tags (e.g. the tags in the root of the dataset), and all tags inside the Shared Functional Groups, plus the tags specific to the given image from the Per Frame Functional Groups. In the [documentation](https://fo-dicom.github.io/dev/v5/usage/raw_pixel_data.html) you can see how to get the frame-specific pixel data which you have to set as `PixelData`. 
You also may have to adapt some tags, for example giving each image a unique SOPInstanceUID.
You can then create a `DicomFile` for each `DicomDataset` and save it.

Why do you need this in the first place? Maybe there is an easier way to achieve what you need?

**Reply by barna21** [reply]
First of all, thank you so much for your reply!

I think it's about EchancedCT. The DICOM files I am talking about are produced by DICOM CT Siemens. And the third parties who are making the CT images don't wan to change the DICOM file format just for us. But I need these single frames files because the conturing system (Phillips Pinnacle) doesn't accept DICOM multi-frame.

Do you think you could share an example? Is there a template in the documentation?

**Reply by mrbean-bremen** [reply]
I don't think there are any templates in the documentation. I may see if I can put together some example, though not today - this may take a bit.
Another possibility would be to use some command line tool that already does this. There is `dcuncat` from David Clunies `dicom3tools` (you can get Windows executables for example [here](https://github.com/QIICR/dicom3tools/releases)), or [gdcmtar](http://gdcm.sourceforge.net/html/gdcmtar.html), which is part of GDCM.

**Comment by mrbean-bremen** [comment, marked as answer]
Just for reference - there is also a [tool based on fo-dicom](https://github.com/RadOncSys/dcmM2S) that does this conversion.

**Reply by barna21** [reply]
Thanks!
